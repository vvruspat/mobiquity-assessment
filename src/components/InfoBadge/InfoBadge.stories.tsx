import { ComponentStory, ComponentMeta } from "@storybook/react";

import { InfoBadge } from "./InfoBadge";
import "../../index.css";

export default {
  title: "Components/InfoBadge",
  component: InfoBadge,
} as ComponentMeta<typeof InfoBadge>;

const Template: ComponentStory<typeof InfoBadge> = (args) => (
  <InfoBadge {...args} />
);

export const InfoBadgeTemplate = Template.bind({});

InfoBadgeTemplate.args = {
  title: "Time",
  content: "1:37'33.584",
};
