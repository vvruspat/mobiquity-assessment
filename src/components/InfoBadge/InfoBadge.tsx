import { Badge } from "../Badge";
import { Caption } from "../Caption";
import { Text } from "../Text";

type InfoBadgeProps = {
  title: string;
  content: string;
};

export const InfoBadge = ({ title, content }: InfoBadgeProps) => {
  return (
    <Badge badgeStyle="info" size="l" align="left">
      <Caption>{title}</Caption>
      <Text size="m" weight="regular">
        {content}
      </Text>
    </Badge>
  );
};
