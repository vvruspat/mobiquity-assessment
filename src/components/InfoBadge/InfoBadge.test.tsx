import { render, screen } from "@testing-library/react";
import { InfoBadge } from "./InfoBadge";

test("renders info badge", () => {
  render(<InfoBadge title="Info badge title" content="Info badge content" />);

  const el = screen.getByTestId("badge");
  expect(el).toHaveTextContent("Info badge content");
  expect(el).toMatchSnapshot();

  const caption = screen.getByTestId("caption");
  expect(caption).toBeInTheDocument();
  expect(caption).toHaveTextContent("Info badge title");
});
