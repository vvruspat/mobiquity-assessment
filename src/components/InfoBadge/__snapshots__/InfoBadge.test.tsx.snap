// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders info badge 1`] = `
<div
  class="badge badge_l badge_style_info badge_align_left"
  data-testid="badge"
>
  <h6
    class="caption"
    data-testid="caption"
  >
    <span
      class="text size_s regular"
      data-testid="text"
    >
      Info badge title
    </span>
  </h6>
  <span
    class="text size_m regular"
    data-testid="text"
  >
    Info badge content
  </span>
</div>
`;
